kratos/includes/properties.h

```diff
@@ -297,24 +297,76 @@ class Properties : public IndexedObject
         }
     }
 
-    /*
-    Method to add Accessors to properties
-    */
     template<class TVariableType>
+    void SetValue(TVariableType const& rV, typename TVariableType::Type const& rValue)
+    {
+        mData.SetValue(rV, rValue);
+    }
+
+    bool HasVariables() const
+    {
+        return !mData.IsEmpty();
+    }
+
+    /**
+     * @brief Set the Accessor object
+     * This method sets a variable-accessor pair in current properties accessor container
+     * @tparam TVariableType The variable type
+     * @param rVariable Variable to which the accessor will refer to
+     * @param pAccessor Pointer to the accessor instance
+     */
+    template <class TVariableType>
     void SetAccessor(const TVariableType& rVariable, AccessorPointerType pAccessor)
     {
         mAccessors.emplace(rVariable.Key(), std::move(pAccessor));
     }
 
-    template<class TVariableType>
-    void SetValue(TVariableType const& rV, typename TVariableType::Type const& rValue)
+    /**
+     * @brief Get the Accessor object
+     * If exists, this method returns a pointer to the requested variable accessor
+     * If doesn't exist, the method throws an error
+     * @tparam TVariableType The variable type
+     * @param rVariable Variable to which the accessor refers to
+     * @return AccessorPointerType& Pointer to the requested variable accessor
+     */
+    template <class TVariableType>
+    Accessor& GetAccessor(const TVariableType& rVariable)
     {
-        mData.SetValue(rV, rValue);
+        auto it_value = mAccessors.find(rVariable.Key());
+        KRATOS_ERROR_IF(it_value == mAccessors.end())
```

Comment:
i wonder if we shouldn't do this check only in debug mode for efficiency. 
This is the same as not verifying that a given [i] entry exists when we access to a vector, and it is something that we normally accept for performance reasons

Reply:
I agree with this observation. This is one of the things we should discuss.

Reply:
Nasty thing is that this would result in a meaningless error...

Reply:
On the other hand, the `Element::Check` method can do the checks to identify (like always) whether required acessors are performed, so we can leave this check only in debug. I'm just suggesting only.